addons/l10n_pe_website_sale/models/res_country_state.py

```diff
@@ -0,0 +1,10 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class CountryState(models.Model):
+    _inherit = "res.country.state"
+
+    def get_website_sale_cities(self):
+        return self.env["res.city"].sudo().search([("state_id", "=", self.id)])
```

Comment:
The benefit is that this method is reusable as it is being called on a [controller](https://github.com/odoo/odoo/blob/303b81fd7acb8dfe678010c3bbb4fab310dbe021/addons/l10n_pe_website_sale/controllers/main.py#L86) that does an rpc already and also [here](https://github.com/odoo/odoo/blob/303b81fd7acb8dfe678010c3bbb4fab310dbe021/addons/l10n_pe_website_sale/controllers/main.py#L60).

Reply:
Ok let me see if I understand, feel free to correct me if I am not right.

Are the 2 extra python files you talk about `res_city.py` and `res_country_state`? The ones that contain the public methods that do a simple search and that should be private?

If so, you want me to delete those and on `main.py` do something like this:
`states = request.env["res.city"].sudo().search([("state_id", "=", self.id)])`
`if state:`
`    to_include["state"] = state`
`    to_include["state_cities"] = states`


Instead of this?
`if state:`
`    to_include["state"] = state`
`    to_include["state_cities"] = state.get_website_sale_cities()`

Reply:
@AndreaGeraldo I think you should be able to call the method search immediately from the js.  There is no need to add an extra controller or to add a public method.  (the less we have to open security-wise, the better, even if it is just states and cities)